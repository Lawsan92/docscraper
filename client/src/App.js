import React, { useState, useEffect } from 'react';
const axios = require('axios');
import '../dist/styles.scss';

export const Landing = ({ setLanding }) => {
  return (
    <div className='scraper_landing' onClick={() => {setLanding(false)}}>
      <div className='scraper_landing-logo'>
        <img src='https://res.cloudinary.com/darp0mj9i/image/upload/v1681251277/Screen_Shot_2023-04-11_at_17.07.44_haqpxe.jpg'style={{height: '40vh', borderRadius: '50%'}}/>
        <h1 style={{color: '#ffce30'}}>DocuScraper</h1>
      </div>
    </div>
  )
}

export const Options = ({ optionsClicked, handleClick, setOptions, configureParams, grepParams }) => {

  const [isChecked, toggleRadio] = useState(false);

  const handleRadio = (param, key) => {
    toggleRadio((prevState )=> ({...isChecked, [key]:! prevState[key]}));
    if (!isChecked[key]) {
      if (param.slice(0, 4) === 'sort') {
        configureParams(currentState => {
          return {
            ...currentState,
            sort: {
              ...currentState.sort,
              [param.slice(5)]: true
            }
          }
        });
      } else {
        // configureParams({...grepParams, [param]: param});
        configureParams(currentState => {
          return {
            ...currentState,
            param: {
              ...currentState.param,
              [param]: true
            }
          }
        });
      }
    } else {
      if (param.slice(0, 4) === 'sort') {
        configureParams(currentState => {
          return {
            ...currentState,
            sort: {
              ...currentState.sort,
              [param.slice(5)]: false
            }
          }
        });

      } else {
        configureParams(currentState => {
          return {
            ...currentState,
            param: {
              ...currentState.param,
              [param]: false
            }
          }
        });
      }
    }
  }

  if (!optionsClicked) {
    return (
      <div className='scraper_options-closed' onClick={handleClick}>
        Options
      </div>
    )
  }

  return (
    <div className='scraper_options-open'>
    <ul style={{listStyle: 'none'}}>
      <li>
        <label>
          <input type='radio' value='email' data-id='1' checked={isChecked['1']} onClick={(e) => { handleRadio(e.target.value, e.target['dataset']['id'])}}/>
          email
        </label>
      </li>
      <li>
        <label>
          <input type='radio' value='phone number' data-id='2' checked={isChecked['2']} onClick={(e) => { handleRadio(e.target.value,  e.target['dataset']['id'])}} />
          phone number
        </label>
      </li>
      <li>
        <label>
          <input type='radio' value='ip address' data-id='3'checked={isChecked['3']} onClick={(e) => { handleRadio(e.target.value,  e.target['dataset']['id'])}} />
          ip address
        </label>
      </li>
      <li>
        <label>
          <input type='radio' value='line numbers' data-id='4' checked={isChecked['4']} onClick={(e) => {handleRadio(e.target.value,  e.target['dataset']['id'])}} />
          line numbers
        </label>
      </li>
      <li>
        <label>
          <input type='radio' value='sort-alphabet' data-id='5' checked={isChecked['5']} onClick={(e) => {handleRadio(e.target.value,  e.target['dataset']['id'])}} />
          sort? (alphabet)
        </label>
      </li>
    </ul>
    <button onClick={handleClick}>x</button>
  </div>
  );
};

export const App = () => {

  // STATE
  const [file, getFile] = useState('');
  const [grepData, getGrep] = useState('');
  const [optionsClicked, openOptions] = useState(false);
  const [doc, getDoc] = useState('');
  const [grepParams, configureParams] = useState({
    sort: {

    }
  });
  const [landing, setLanding] = useState(true);

  useEffect(() => {
    fetchFile();
  }, [])

  // METHODS
  const fetchFile = async () => {
    // const fetchRequest = await fetch('./getFiles');
    // const fileData = await fetchRequest.json();
    axios({
      url: './getFiles',
      method: 'get'
    })
    .then((response) => {
      // console.log('response:', response.data);
      const fileData = response.data;
      // getFile(fileData);
    })
  }

  const uploadFile = (e) => {
    let file = e.target.files[0];
    const reader = new FileReader();
    reader.onload = (e) => {
      let fileContent = e.target.result;
      getFile(fileContent);
    };
    reader.readAsText(file);
  }

  const grepFile = (file) => {
    const reader = new FileReader();
    reader.onload = (e) => {
      let fileContent = e.target.result;
      // console.log('fileContent:', fileContent, typeof fileContent);
      // console.log('grepParams:', grepParams)
      axios({
        url: './grepFiles',
        method: 'post',
        data: {data: fileContent, options: grepParams}
      })
      .then((response) => {
        console.log('response:', response.data);
        getGrep(response.data);
      })
    };
    reader.readAsText(file);
  }

  const handleClick = () => {
    openOptions(prevState => !prevState);
  };

  const setOptions = (param) => {
    configureParams({...grepParams, [param]: param});
  }

  if (landing) {
    return <Landing setLanding={setLanding}/>
  } else {
    return (
      <div className='scraper_container'>
        <div className='scraper_pages'>
          <div className='scraper_before-page'>
            <p>{file}</p>
          </div>
          <div className='scraper_after-page'>
            <p style={{color: 'red'}}>{grepData}</p>
          </div>
        </div>
        <div className='scraper_dash'>
          <input type="file" name="file" onChange={(e) => { uploadFile(e); getDoc(e.target.files[0]);/*grepFile(e.target.files[0]);*/}}/>
          <button onClick={() => {grepFile(doc); configureParams({})}}>Filter</button>
          <Options optionsClicked={optionsClicked} handleClick={handleClick} setOptions={setOptions} configureParams={configureParams} grepParams={grepParams}/>
        </div>
      </div>
    );
  }
}

// REFERENCES
// https://www.pluralsight.com/guides/how-to-access-custom-attributes-from-aevent-object-in-react
// https://bobbyhadz.com/blog/react-remove-key-from-state-object
// https://bobbyhadz.com/blog/react-update-nested-state-object-property