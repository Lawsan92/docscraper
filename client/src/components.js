import React from 'react';
import App from './App.js';
import Landing from './Landing.js';
import Options from './Options.js';

export {
  App,
  Landing,
  Options
}