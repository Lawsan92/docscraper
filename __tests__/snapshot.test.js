import renderer from 'react-test-renderer';
import { Landing, Options } from '../client/src/components.js';
import App from '../client/src/App.js';
import React from 'react';
import Component from '../component.js';
import { isElement } from 'react-dom/test-utils';
const log = process.argv[process.argv.length - 1] === '-log'; // jest __tests__/api.test.js -log

log && console.log('process.argv:', process.argv, 'process.argv[process.argv.length - 1:', process.argv[process.argv.length - 1]);

describe('DocuScraper UI', () => {

  describe('<Landing/>', () => {

    test('it should be recognized as a React element', () => {
      log && console.log('<Landing/>:', <Landing/>);
      expect(isElement(<Landing/>)).toBe(true)
    });

    test('it matches snapshot', () => {
      const landing = renderer
        .create(<Landing/>)
        .toJSON();
      log && console.log('landing:', landing);
      expect(isElement(<Landing/>)).toBe(true)
      expect(landing).toMatchSnapshot();
    });

  });

  describe('<App/>', () => {

    test('it should be recognized as a React element', () => {
      log && console.log('<App/>:', <App/>);
      expect(isElement(<App/>)).toBe(true)
    });

    test('it matches snapshot', async () => {
      try {
        const app = renderer
        .create(<App/>)
        .toJSON();
      log && console.log('app:', app);
      expect(app).toMatchSnapshot();
      } catch (e) {
        expect(e).toMatch('error');
      }
    });

  });

  describe('<Options/>', () => {

    test('it should be recognized as a React element', () => {
      log && console.log('<Options/>:', <Options/>);
      expect(isElement(<Options/>)).toBe(true)
    });

    test('it matches snapshot', () => {
      const options = renderer
        .create(<Options/>)
        .toJSON();
      log && console.log('options:', options);
      expect(options).toMatchSnapshot();
    });

  });

  describe('<Component/>', () => {

    test('it should be recognized as a React element', () => {
      log && console.log('<Component/>:', <Component/>);
      expect(isElement(<Component/>)).toBe(true)
    });

    test('it matches snapshot', () => {
      const component = renderer
        .create(<Component/>)
        .toJSON();
      log && console.log('component:', component);
      expect(component).toMatchSnapshot();
    });

  });

});

