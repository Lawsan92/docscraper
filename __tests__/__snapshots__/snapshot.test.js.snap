// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocuScraper UI <App/> it matches snapshot 1`] = `
<div
  className="scraper_landing"
  onClick={[Function]}
>
  <div
    className="scraper_landing-logo"
  >
    <img
      src="https://res.cloudinary.com/darp0mj9i/image/upload/v1681251277/Screen_Shot_2023-04-11_at_17.07.44_haqpxe.jpg"
      style={
        Object {
          "borderRadius": "50%",
          "height": "40vh",
        }
      }
    />
    <h1
      style={
        Object {
          "color": "#ffce30",
        }
      }
    >
      DocuScraper
    </h1>
  </div>
</div>
`;

exports[`DocuScraper UI <Component/> it matches snapshot 1`] = `
<div>
  COMPONENT
</div>
`;

exports[`DocuScraper UI <Landing/> it matches snapshot 1`] = `
<div
  className="scraper_landing"
  onClick={[Function]}
>
  <div
    className="scraper_landing-logo"
  >
    <img
      src="https://res.cloudinary.com/darp0mj9i/image/upload/v1681251277/Screen_Shot_2023-04-11_at_17.07.44_haqpxe.jpg"
      style={
        Object {
          "borderRadius": "50%",
          "height": "40vh",
        }
      }
    />
    <h1
      style={
        Object {
          "color": "#ffce30",
        }
      }
    >
      DocuScraper
    </h1>
  </div>
</div>
`;

exports[`DocuScraper UI <Options/> it matches snapshot 1`] = `
<div
  className="scraper_options-closed"
>
  Options
</div>
`;
